interface/globals.php

```diff
@@ -287,6 +287,14 @@ function undoMagicQuotes($array, $topLevel=true) {
       if ($gl_value == '2') $GLOBALS['sell_non_drug_products'] = 1;
       else if ($gl_value == '3') $GLOBALS['sell_non_drug_products'] = 2;
     }
+    else if ($gl_name == 'gbl_time_zone') {
+      // If value is empty default to old behavior.
+      if ($gl_value) {
+        date_default_timezone_set($gl_value);
+        // Synchronize MySQL time zone with PHP time zone.
+        sqlStatement("SET time_zone = ?", array((new DateTime())->format("P")));
```

Comment:
Definitely a risk in those places since can't control the queries from there. But it's an odd dilemma, though, since need access to the database to collect the global. I think you could populate the global where you do in globals and also update the main mysql connector there. And then should be able to use the populated global in all the other sql connectors since they all connect after running globals.php. This feature has been requested for some time, so would be nice to support. I think can ignore Installer.class.php since that is for installation only and uses native mysqli calls.